packages/docs-reanimated/docs/advanced/useEvent.mdx

```diff
@@ -4,26 +4,86 @@ sidebar_position: 4
 
 # useEvent
 
-import DocsCompatibilityInfo from '../_shared/_docs_compatibility_info.mdx';
+`useEvent` is a low-level hook. It returns event handler that will be called when native event occurs. You can use it to create custom event handler hooks, like `useAnimatedGestureHandler` or `useAnimatedScrollHandler`.
 
-<DocsCompatibilityInfo />
+## Reference
 
-This is low-level hook returning event handler that will be invoked with native events, which should be used in order to create custom event handler hook like `useAnimatedGestureHandler` or `useAnimatedScrollHandler`.
+```js
+function useEventExample() {
+  const dependencies = [{ isWeb: false }];
+  const handlers = {
+    onScroll: (event) => {
+      'worklet';
+      console.log(event);
+    },
+  };
+
+  const { context, doDependenciesDiffer, useWeb } = useHandler(
+    handlers,
+    dependencies
+  );
+
+  // highlight-start
+  const customScrollHandler = useEvent(
+    (event) => {
+      'worklet';
+      const { onScroll } = handlers;
+      if (onScroll && event.eventName.endsWith('onScroll')) {
+        context.eventName = event.eventName + useWeb;
+        onScroll(event);
+      }
+    },
+    ['onScroll'],
+    doDependenciesDiffer
+  );
+  // highlight-end
+
+  return <Animated.ScrollView onScroll={customScrollHandler} />;
+}
+```
+
+<details>
+<summary>Type definitions</summary>
+
+```typescript
+function useEvent<
+  Event extends object,
+  Context extends Record<string, unknown> = never
+>(
+  handler: EventHandler<Event, Context>,
+  eventNames?: string[],
+  rebuild?: boolean
+): EventHandlerProcessed<Event, Context>;
+
+type EventHandler<
+  Event extends object,
+  Context extends Record<string, unknown> = never
+> = (event: ReanimatedEvent<Event>, context?: Context) => void;
+
+type EventHandlerProcessed<
+  Event extends object,
+  Context extends Record<string, unknown> = never
+> = (event: Event, context?: Context) => void;
+```
+
+</details>
 
 ### Arguments
 
-#### `handler` [function]
+#### `handler`
 
-Handler will receive event object with native payload, that can be passed to custom handler hook's worklets.
+Function that receives event object with native payload, that can be passed to custom handler hook's worklets.
 
-- `event` [object] - event object.
+- `event` - event object.
   The payload can differ depending on the type of the event.
 
-#### `eventNames` [Array]
+- `context` <Optional /> - JS object that was used to store some state.
```

Comment:
I don't know if I would include this is as an argument. It is more of a generic field rather than a fully-pledged argument. Also, I wouldn't put it int fron of `eventName` which seems to be more important to the hook. Lastly:
```suggestion
- `context` <Optional /> - JS object that is used to store some state.
```

Reply:
i will remove that then, to avoid confusion